src/Document/Meta.php

```diff
@@ -21,4 +23,22 @@ public function jsonSerialize()
     {
         return $this->data;
     }
+
+    private function validateObject($object)
+    {
+        foreach ($object as $name => $value) {
```

Comment:
What if `$object` implements `Iterator` and the keys and values exposed by `Iterator` interface are different from its JSON representation. We may perhaps do json_encode right in the constructor, then convert it back to an array with json_decode(), and then check the keys of the resulting array.